src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__okrts_behavior.sql

```diff
@@ -0,0 +1,162 @@
+            co.is_retained_year,
+
+            hr.sections_section_number as homeroom_section,
+            hr.teacher_lastfirst as homeroom_teacher_name,
+
+            w.week_start_monday,
+            w.week_end_sunday,
+            w.date_count as days_in_session,
+            w.quarter as term,
+
+            bc.category_type,
+
+            b.dl_said,
+            b.point_value,
+
+            concat(b.staff_last_name, ', ', b.staff_first_name) as entry_staff,
+            case
+                when co.region = 'Miami'
+                then regexp_extract(b.behavior_category, r'^(.*?) \(')
+                else b.behavior
+            end as behavior,
+
+            count(distinct co.student_number) over (
+                partition by w.week_start_monday, co.school_abbreviation
+            ) as school_enrollment_by_week,
+
+            if(co.lep_status, 'ML', 'Not ML') as ml_status,
+            if(co.is_504, 'Has 504', 'No 504') as status_504,
+            if(
+                co.is_self_contained, 'Self-contained', 'Not self-contained'
+            ) as self_contained_status,
+            if(co.spedlep like 'SPED%', 'Has IEP', 'No IEP') as iep_status,
+        from {{ ref("base_powerschool__student_enrollments") }} as co
+        left join
+            {{ ref("base_powerschool__course_enrollments") }} as hr
+            on co.studentid = hr.cc_studentid
+            and co.yearid = hr.cc_yearid
+            and co.schoolid = hr.cc_schoolid
+            and {{ union_dataset_join_clause(left_alias="co", right_alias="hr") }}
+            and hr.cc_course_number = 'HR'
+            and not hr.is_dropped_section
+            and hr.rn_course_number_year = 1
+        inner join
+            {{ ref("int_powerschool__calendar_week") }} as w
+            on co.academic_year = w.academic_year
+            and co.schoolid = w.schoolid
+            and w.week_end_sunday between co.entrydate and co.exitdate
+        left join
+            {{ ref("stg_deanslist__behavior") }} as b
+            on co.student_number = b.student_school_id
+            and b.behavior_date between w.week_start_monday and w.week_end_sunday
+        inner join
+            behavior_categories as bc
+            on co.region = bc.region
+            and b.behavior_category = bc.behavior_category
+        where
+            co.academic_year >= {{ var("current_academic_year") }} - 1
+            and co.rn_year = 1
+            and co.grade_level != 99
+    ),
+
+    behavior_aggregation as (
+        select
+            student_number,
+            academic_year,
+            week_start_monday,
+            category_type,
+            behavior,
+            entry_staff,
+
+            count(distinct dl_said) as behavior_count,
+            sum(point_value) as total_points,
+        from roster
+        group by
+            student_number,
+            academic_year,
+            week_start_monday,
+            category_type,
+            behavior,
+            entry_staff
```

Comment:
double check that you want these in the grouping, would make the counts long by behavior name and staff

Reply:
Yes, that is there because otherwise there's no way to do staff entry analysis. Old data set was ~10m rows, the view is ~900k without aggregating around entry staff, and ~1.7m with it added, so still about 1/5 of the rows of just bringing in the raw behavior data as one row per entry.